src/tabs.ts

```diff
@@ -0,0 +1,258 @@
+        this._openActive();
+    }
+    private _showAccordion() {
+        if(this.animation) {
+            let style = document.createElement("style");
+            style.innerHTML = `${animationCSS(
+                '.animation-open', '.animation-close',
+                this._animation_styles, this.animationDur)}`
+            this.appendChild(style);
+        }
+        let title = this.querySelectorAll("[slot=tab-title]");
+        let body = this.querySelectorAll("[slot=tab-body]");
+        for (let i = 0;i < Math.min(title.length, body.length); i++) {
+            let div = document.createElement("div");
+            div.slot = "acc";
+            div.appendChild(title[i]);
+            div.appendChild(body[i]);
+            this.appendChild(div);
+        }
+    }
+    private _showRC() {
+        let acc = this.querySelectorAll("[slot=acc]");
+        for(let j of acc) {
+            this.appendChild(j.children[1]);
+            this.appendChild(j.children[0]);
+            j.remove();
+        }
+        let styles = this.querySelectorAll("style");
+        for (let s of styles)
+            s.remove();
+    }
+    private _titleClicked(e:Event) {
+        let title = this.querySelectorAll("[slot=tab-title]");
+        for(let ix = 0;ix < title.length; ix++)
+            if(title[ix] == e.target && this._prev_ix != ix)
+                this.show(ix);
+    }
+    private _animateStyles() {
+        if(!this.animation)
+            return;
+        return html`
+<style>
+    ${animationCSS(
+        '::slotted(.animation-open)',
+        '::slotted(.animation-close)',
+        this._animation_styles,
+        this.animationDur)}
+</styel>`
+    }
+    private _show(ix:number, prev_ix:number) {
+        let title = this.querySelectorAll("[slot=tab-title]");
+        let body = this.querySelectorAll("[slot=tab-body]");
+        if(prev_ix != null) {
+            this.dispatchEvent(
+                new CustomEvent("hide", { detail: { ix } })
+            );
+            if(this.animationDur) {
+                body[prev_ix]?.classList.remove("animation-open");
+                body[prev_ix]?.classList.add("animation-close");
+            }
+            if(body[prev_ix])
+                (body[prev_ix] as HTMLElement).style.display = 'none';
+            body[prev_ix]?.classList.remove(this.activecls)
+            title[prev_ix]?.classList.remove(this.activecls);
+        }
+        this.dispatchEvent(
+            new CustomEvent("show", { detail: { ix } })
+        );
+        title[ix]?.classList.add(this.activecls);
+        body[ix]?.classList.add(this.activecls);
+        if(this.animationDur) {
+            body[ix]?.classList.remove("animation-close");
+            body[ix]?.classList.add("animation-open");
+        }
+        if(body[ix])
+            (body[ix] as HTMLElement).style.display = 'block';
+    }
+    show(ix:number) {
+        this.dispatchEvent(
+            new CustomEvent("change", { detail: { ix } })
+        );
```

Comment:
It will be good to call the even dispatch after the real change has happened. So the best place would at the bottom of this function